src/Paramore.Brighter/Transforms/Transformers/CloudEventsTransformer.cs

```diff
@@ -25,48 +23,71 @@ public class CloudEventsTransformer : IAmAMessageTransform
 {
     private Uri? _source;
     private string? _type;
-    private string? _datacontenttype;
-    private Uri? _dataschema;
+    private string? _specVersion;
+    private string? _dataContentType;
+    private Uri? _dataSchema;
     private string? _subject;
-    
+
     /// <summary>
     /// Gets or sets the context. Usually the context is given to you by the pipeline and you do not need to set this
     /// </summary>
     /// <value>The context.</value>
     public IRequestContext? Context { get; set; }
-    
+
     public void Dispose()
     {
         //no op as we have no unmanaged resources
     }
 
-    public void InitializeWrapFromAttributeParams(params object[] initializerList)
+    public void InitializeWrapFromAttributeParams(params object?[] initializerList)
     {
-        
-        _source = initializerList.ElementAtOrDefault(0) == null ? null : (Uri) initializerList.ElementAtOrDefault(0)!; 
-        _type = initializerList.ElementAtOrDefault(1)?.ToString(); 
-        _datacontenttype = initializerList.ElementAtOrDefault(2)?.ToString();
-        _dataschema = initializerList.ElementAtOrDefault(3) == null ? null : (Uri)initializerList.ElementAtOrDefault(3)!; 
-        _subject = initializerList.ElementAtOrDefault(4)?.ToString();
+        if (initializerList[0] is string source)
+        {
+            _source = new Uri(source, UriKind.RelativeOrAbsolute);
+        }
+
+        if (initializerList[1] is string type)
+        {
+            _type = type;
+        }
+
+        if (initializerList[2] is string specVersion)
+        {
+            _specVersion = specVersion;
+        }
+
+        if (initializerList[3] is string dataContentType)
```

Comment:
The if syntax is clearer, but all the brackets spread it out so much it's hard to read the logic. I think this was better before